plonky2/Cargo.toml

```diff
@@ -11,45 +11,46 @@ edition = "2021"
 default-run = "generate_constants"
 
 [features]
-default = ["parallel", "rand", "rand_chacha", "timing", "gate_testing"]
-parallel = ["maybe_rayon/parallel"]
-rand = ["dep:rand", "plonky2_field/rand"]
-gate_testing = ["rand"]
-rand_chacha = ["dep:rand_chacha"]
-timing = []
+default = ["gate_testing", "parallel", "rand_chacha", "std", "timing"]
+gate_testing = []
+parallel = ["hashbrown/rayon", "maybe_rayon/parallel"]
+std = ["anyhow/std", "rand/std"]
+timing = ["std"]
 
 [dependencies]
-plonky2_field = { path = "../field" }
-plonky2_util = { path = "../util" }
-log = "0.4.14"
-itertools = "0.10.0"
-num = { version = "0.4", features = [ "rand" ] }
-rand = { version = "0.8.4", optional = true }
-rand_chacha = { version = "0.3.1", optional = true }
-maybe_rayon = { path = "../maybe_rayon" }
-unroll = "0.1.5"
-anyhow = "1.0.40"
-serde = { version = "1.0", features = ["derive"] }
-serde_cbor = "0.11.1"
-keccak-hash = "0.8.0"
-static_assertions = "1.1.0"
+anyhow = { version = "1.0.40", default-features = false }
+derivative = { version = "2.2.0", default-features = false, features = ["use_core"] }
```

Comment:
It seems like we're not making that much use of derivative's features/flexibility? We generally like to keep our dependencies lean, so would prefer to avoid this if the benefits are minor.

Reply:
I won't go strongly on this one either way except to say that the standard Rust `derive` does not have the correct semantics and that `derivative` is a way to correct this. I didn't replace every `derive` with `derivative` in the crate for minimality reasons, but I think that's the correct thing to do in general. If you want to move to that, I'm happy to do another PR to fix this, but I'm alright removing `derivative`.

Reply:
Hm I'd like to learn more about the concerns with Rust derives, but I'd say let's avoid derivative for now and discuss later.

There are all minor, but I just prefer standard-Rust things by default since
- It's best for build times
- Very little chance of bugs
- No learning curve, since everyone is familiar with them

Reply:
Yeah I can elaborate in a GitHub issue, but it's non-crucial right now so I removed the dependency.